rust/libnewsboat/src/utils.rs

```diff
@@ -1037,6 +1037,33 @@ pub fn locale_to_utf8(text: &[u8]) -> String {
     String::from_utf8(converted).expect("convert_text() returned a non-UTF-8 string")
 }
 
+/// Parses a string in the form `author@example.com (Example Author)`
+/// into a pair of (author, email).
+/// If the pattern cannot be matched or the name is empty, the input
+/// is returned completely as author with empty email.
+///
+/// Invalid data (outside of utf-8) is dropped
+pub fn parse_rss_author_email(text: &[u8]) -> (String, String) {
+    let re = regex::bytes::Regex::new(r#"(.+@.+) \((.+)\)"#)
```

Comment:
I think the regex should start with `^` and end with `$` to avoid matching a part of the input. For example, "@ (Foo) Bar" will currently be parsed as `("@", "Foo")`, whereas it should fail to match the regex at all.

Reply:
Indeed, good catch. Fixed